### PR TITLE
Bring compatibility page up to date

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
Thanks so much for the contribution!
To make reviewing this PR a bit easier, please fill out answers to the following questions.

### What was the end-user problem that led to this PR?

The problem was that this page is out of date.

### What is your fix for the problem, implemented in this PR?

My fix to document our current policy. I don't think we need to document all past compatibility issues here.

### CODE REVIEW

File: source/compatibility.html.haml [hunk 1]
```diff
@@ -5,79 +5,42 @@ description: Ruby and RubyGems requirements needed for Bundler compatibility
...
-        %li Requires Bundler 1.0.10 or greater. Ruby <b>1.9.1</b> is not supported.
-
-  .bullet
-    .description
-      Bundler <b>2.0</b> requires a minimum Ruby version of <b>2.3.0</b>.
-      Bundler <b>1.0</b> requires a minimum Ruby version of <b>1.8.7</b>.
-
-%h2 Bundler compatibility with RubyGems
+%p.contents
+  The latest bundler release should always support, at the very least, all
```
Comment (olleolleolle) [inline source/compatibility.html.haml:9]: Capitalize Bundler

File: source/compatibility.html.haml [hunk 2]
```diff
@@ -5,79 +5,42 @@ description: Ruby and RubyGems requirements needed for Bundler compatibility
...
+%h2 Bundler compatibility with RubyGems
 
-  .bullet
-    .description
-      RubyGems 1.6 and 1.7
-    .notes
-      %ul
-        %li Requires Bundler version 1.0.11.
+%p.contents
+  The latest bundler release should always support all RubyGems versions higher
```
Comment (olleolleolle) [inline source/compatibility.html.haml:29]: Perhaps capitalize Bundler here, too?

**Review by olleolleolle — APPROVED** [review]
Super upgrade!

**Comment by deivid-rodriguez** [comment]
Thanks for having a look @olleolleolle, fixed!

**Pull request merged**